cmd/senpai/main.go

```diff
@@ -51,16 +53,23 @@ func main() {
 	lastNetID, lastBuffer := getLastBuffer()
 	app.SwitchToBuffer(lastNetID, lastBuffer)
 
+	// Write last buffer on close
+	sigCh := make(chan os.Signal, 1)
+	signal.Notify(sigCh, syscall.SIGINT, syscall.SIGTERM, syscall.SIGHUP)
+
+	go func() {
+		for sig := range sigCh {
+			switch sig {
+			case syscall.SIGHUP, syscall.SIGINT, syscall.SIGTERM:
+				app.Close()
+				writeLastBuffer(app)
+			}
+		}
```

Comment:
No need for a for-loop here.

IIUC sigCh only receives the signals we asked for, so you can replace it with `<-sigCh`.